titus-supplementary-component/tasks-publisher/src/main/java/com/netflix/titus/supplementary/taskspublisher/TaskDocument.java

```diff
@@ -564,6 +568,18 @@ private static TitusTaskState toV2TaskState(TaskStatus taskStatus) {
         }
     }
 
+    @VisibleForTesting
+    static Map<String, String> sanitizeEnvMap(Map<String, String> env) {
+        if (env == null) {
+            return Collections.emptyMap();
+        }
+        return env.keySet().stream().filter(TaskDocument::isSafe).collect(Collectors.toMap(k -> k, env::get));
```

Comment:
Might be useful to have a log line or a metric to know how often this is happening.